### ISSUE TITLE
`native` version fails to install on Node v20

### ISSUE THREAD

**Issue by danielrozenberg** [jridgewell/safe-decode-uri-component#6]
(I have a solution, but couldn't figure out the branch situation between native/non-native so I wasn't sure how to send out a PR - solution below)

I'm getting the following error message when trying to run `npm install safe-decode-uri-component@1.2.1-native` (or [full log](https://gist.github.com/danielrozenberg/a9b5699656446c3acd7e164f09d066b0)):

```
npm ERR! gyp info using node-gyp@5.1.1
npm ERR! gyp info using node@20.9.0 | linux | x64
npm ERR! gyp info find Python using Python version 3.11.5 found at "/usr/bin/python"
npm ERR! gyp ERR! UNCAUGHT EXCEPTION 
npm ERR! gyp ERR! stack TypeError: Cannot assign to read only property 'cflags' of object '#<Object>'
npm ERR! gyp ERR! stack     at createConfigFile (/usr/local/google/home/rodaniel/error-tracker/node_modules/node-gyp/lib/configure.js:118:21)
npm ERR! gyp ERR! stack     at /usr/local/google/home/rodaniel/error-tracker/node_modules/node-gyp/lib/configure.js:85:9
npm ERR! gyp ERR! stack     at /usr/local/google/home/rodaniel/error-tracker/node_modules/mkdirp/index.js:30:20
npm ERR! gyp ERR! stack     at FSReqCallback.oncomplete (node:fs:189:23)
```

Obviously the `@1.2.1` (not-native) version works, and running `npm install --ignore-scripts safe-decode-uri-component@1.2.1-native` installs successfully but fails to actually run since it doesn't get built.

**Solution**
I tried installing with `--ignore-scripts`, then updating all the dependencies and devDependencies in `node_modules/safe-decode-uri-component/package.json` and running `npm install` inside that directory, and it all worked. So it looks like simply updating the deps and releasing a new version will fix this issue

Hi Justin :wave:

**Comment by jridgewell** [comment]
Hi Daniel!

You just need to open a PR against the native branch. There's a little switcher at the top of the PR page that will allow you to switch. Or, just open the PR against main and I'll switch it.

**Comment by danielrozenberg** [comment]
The `native` looks out of date though (package.json has a version of 1.1.4 on it)
I'll send a PR based on the `1.2.1-native` tag and open it against the `native` branch, you can probably figure it out from there :)

**Comment by jridgewell** [comment]
Sorry, I forgot I switch to the `n-api` branch (instead of using V8 internals, it uses Node's public C++ API). The updated deps should be published as part of `1.2.2-native`

**Comment by danielrozenberg** [comment]
Awesome, thanks!!
